apps/frontend/static_src/js/main.js

```diff
@@ -1 +1,70 @@
 import { Dropdown } from 'bootstrap';
+
+const searchInput = document.querySelector('[data-search-input]')
+const searchIconButton = document.querySelector('[data-search-icon-button]')
+
+const onSearchInputChange = async (event) => {
+    const query = event.target.value
+    try {
+        const res = await fetch(window.location.origin + '/search_json/?query=' + query)
+        const data = await res.json()
+        injectResultsInHTML(data)
+    } catch(err) {
+        console.log(err)
+        return {"error": err}
```

Comment:
Unless I’m misunderstanding the code, this return value doesn’t seem to be used anywhere. Could we instead use `window.alert` to show an error message for end users?

Reply:
Yeah you are right. Initially I didn't know what to do if error occurs, therefore I just logged it out to debug.